Backpack/Snackbar/Classes/BPKSnackbar.m

```diff
@@ -0,0 +1,370 @@
+        [self.snackbarView addSubview:self.actionButton];
+        [self.snackbarView addSubview:self.titleLabel];
+        [self.snackbarView addSubview:self.leftIconContainer];
+
+        [NSLayoutConstraint activateConstraints:@[
+            [self.snackbarView.widthAnchor constraintEqualToAnchor:self.widthAnchor],
+            [self.snackbarView.heightAnchor constraintEqualToAnchor:self.heightAnchor],
+            [self.leftIconContainer.leadingAnchor constraintEqualToAnchor:self.snackbarView.leadingAnchor
+                                                                 constant:BPKSpacingLg],
+            [self.leftIconContainer.centerYAnchor constraintEqualToAnchor:self.snackbarView.centerYAnchor],
+            [self.titleLabel.centerYAnchor constraintEqualToAnchor:self.snackbarView.centerYAnchor],
+            [self.snackbarView.trailingAnchor constraintEqualToAnchor:self.actionButton.trailingAnchor
+                                                             constant:BPKSpacingLg],
+            [self.actionButton.centerYAnchor constraintEqualToAnchor:self.snackbarView.centerYAnchor]
+        ]];
+
+        self.snackbarViewTopConstraint = [self.snackbarView.topAnchor constraintEqualToAnchor:self.topAnchor];
+        self.snackbarViewTopConstraint.active = YES;
+
+        self.labelButtonSpacing =
+            [self.actionButton.leadingAnchor constraintGreaterThanOrEqualToAnchor:self.titleLabel.trailingAnchor
+                                                                         constant:BPKSpacingBase];
+        self.labelButtonSpacing.active = YES;
+
+        self.labelLeadingSpacing =
+            [self.titleLabel.leadingAnchor constraintEqualToAnchor:self.leftIconContainer.trailingAnchor
+                                                          constant:BPKSpacingMd];
+        self.labelLeadingSpacing.active = YES;
+
+        self.snackbarView.accessibilityIdentifier = SnackbarAccessibilityIdentifier;
+
+        self.keyboardHeight = 0.0f;
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(keyboardWillChange:)
+                                                     name:UIKeyboardWillChangeFrameNotification
+                                                   object:nil];
+    }
+
+    return self;
+}
+
+- (void)setUpSnackbarWithTitle:(NSString *)title
+                   buttonTitle:(NSString *_Nullable)buttonTitle
+                    buttonIcon:(UIImage *_Nullable)buttonIcon
+                      leftIcon:(UIImage *_Nullable)leftIcon
+                      duration:(BPKSnackbarDuration)duration
+                viewController:(UIViewController *)viewController
+                      delegate:(id<BPKSnackbarProtocol> _Nullable)delegate {
+    self.titleLabel.text = title;
+    self.delegate = delegate;
+    self.snackbarView.backgroundColor = BPKColor.skyBlueShade02;
+    self.parentViewController = viewController;
```

Comment:
Could we maybe rename this so that we don't get confused with this:
https://developer.apple.com/documentation/uikit/uiviewcontroller/1621362-parentviewcontroller